boundary/analyze.py

```diff
@@ -263,6 +266,31 @@ def func_class_arithmetics(fns):
     fns.tainted = (fns.border | fns.insider | fns.sidecar) & fns.in_vmlinux
     fns.und = (fns.sched_outsider - fns.outsider_opt) | fns.border | fns.sidecar
 
+def get_func_decl_strs(signatures, fmt):
+    """Generate function declaration strings. If both strong and weak
+    symbol exist, keep only one.
+    """
+    decl_strs = set()
+    local_syms = set()
+
+    for fn in signatures:
+        (name, file) = fn
+        s = fmt.format(**decls[fn])
+
+        if file != global_fn_dict.get(name):
```

Comment:
Why not just `if name not in global_fn_dict:`

Reply:
It can't handle the following case:

(hello, kernel/sched/core.c)   ---> global symbol
(hello, kernel/sched/fair.c)     ---> static symbol